test/unit/test_id_storage.c

```diff
@@ -0,0 +1,187 @@
+#include <r_util.h>
+#include "minunit.h"
+
+static bool test_r_id_storage_toomany(void) {
+	bool r;
+	ut32 id0, id1, id2;
+	RIDStorage *ids = r_id_storage_new (20, 22);
+	r = r_id_storage_add (ids, "aaa", &id0);
+	mu_assert ("id0", r);
+	r = r_id_storage_add (ids, "bbb", &id1);
+	mu_assert ("id1", r);
+	r = r_id_storage_add (ids, "ccc", &id2);
+	mu_assert ("id2", !r);
+	mu_end;
+}
+
+static bool test_r_id_storage_wrong(void) {
+	bool r;
+	ut32 id0, id1;
+	RIDStorage *ids = r_id_storage_new (20, 10);
+	r = r_id_storage_add (ids, "aaa", &id0);
+	mu_assert ("id0", !r);
+	mu_end;
+}
+
+static bool test_r_id_storage_prev_next_eq_0(void) {
+	// test if next reverts prev
+	bool r;
+	ut32 id0, id1, id2;
+	RIDStorage *ids = r_id_storage_new (1, 15);
+	r = r_id_storage_add (ids, "aaa", &id0);
+	mu_assert ("id0", r);
+	mu_assert ("id0", id0 == 1);
+	r = r_id_storage_add (ids, "bbb", &id1);
+	mu_assert ("id1", r);
+	mu_assert ("id1", id1 == 2);
+	r = r_id_storage_add (ids, "ccc", &id2);
+	mu_assert ("id2", r);
+	mu_assert ("id2", id2 == 3);
+	ut32 id = id1;
+	r_id_storage_get_prev (ids, &id);
+	r_id_storage_get_next (ids, &id);
+	r_id_storage_free (ids);
+	
+	mu_assert_eq (id, id1, "r_id_storage_{next/prev} reversal 0");
+	mu_end;
+}
+
+static bool test_r_id_storage_prev_next_eq_1(void) {
+	// test if next reverts prev (modulo wrap 1)
+	bool r;
+	ut32 id, id0, id1, id2;
+	RIDStorage *ids = r_id_storage_new (0, 15);
+	r_id_storage_add (ids, "bbb", &id0);
+	r_id_storage_add (ids, "aaa", &id1);
+	r_id_storage_add (ids, "ccc", &id2);
+	id = id0;
+	r = r_id_storage_get_prev (ids, &id);
+	mu_assert ("get_prev(0) doesnt exist", !r);
+	ut32 n = id;
+	id = id1;
+	r = r_id_storage_get_prev (ids, &id);
+	mu_assert ("get_prev(1) must exist", r);
+	r = r_id_storage_get_next (ids, &id);
+	mu_assert ("get_prev(1) must exist", r);
+	r_id_storage_free (ids);
+	
+	mu_assert_eq (id, id1, "r_id_storage_{next/prev} reversal 1");
+	mu_end;
+}
+
+static bool test_r_id_storage_prev_next_min(void) {
+	// test if next reverts prev (modulo wrap 1)
+	bool r;
+	ut32 id, id0, id1, id2;
+	RIDStorage *ids = r_id_storage_new (3, 15);
+	r_id_storage_add (ids, "bbb", &id0);
+	r_id_storage_add (ids, "aaa", &id1);
+	r_id_storage_add (ids, "ccc", &id2);
+	id = id0;
+	r = r_id_storage_get_prev (ids, &id);
+	mu_assert ("get_prev(0) doesnt exist", !r);
+	ut32 n = id;
+	id = id1;
+	r = r_id_storage_get_prev (ids, &id);
+	mu_assert ("get_prev(1) must exist", r);
+	r = r_id_storage_get_next (ids, &id);
+	mu_assert ("get_prev(1) must exist", r);
+	r_id_storage_free (ids);
+	mu_assert_eq (id, id1, "r_id_storage_{next/prev} reversal 1");
+	mu_end;
+}
+
+static bool test_r_id_storage_empty(void) {
+	// test if next reverts prev (modulo wrap 2)
+	ut32 id, _id;
+	RIDStorage *ids = r_id_storage_new (0, 15);
+	bool r = r_id_storage_get_prev (ids, &_id);
```

Comment:
`_id` should probably be initialized.

Reply:
It doesn't really matter in this case because it won't probably be used since `ids` is empty, but..